core/conn/spj_init/src/main/resources/init_spj.sql

```diff
@@ -0,0 +1,156 @@
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE HELP TO PUBLIC;
+      
+      CREATE PROCEDURE PUT (
+      IN FILEDATA VARCHAR(102400) CHARACTER SET ISO88591,
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      IN CREATEFLAG INTEGER)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.put(java.lang.String,java.lang.String,int)'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE PUT TO PUBLIC;
+      
+      CREATE PROCEDURE LS (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      OUT FILENAMES VARCHAR(10240) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.ls(java.lang.String,java.lang.String[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE LS TO PUBLIC;
+      
+      CREATE PROCEDURE LSALL (
+      OUT FILENAMES VARCHAR(10240) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.lsAll(java.lang.String[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE LSALL TO PUBLIC;
+      
+      CREATE PROCEDURE RM (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.rm(java.lang.String)'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE RM TO PUBLIC;
+      
+      CREATE PROCEDURE RMREX (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      OUT FILENAMES VARCHAR(10240) CHARACTER SET ISO88591)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.rmRex(java.lang.String, java.lang.String[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE RMREX TO PUBLIC;
+
+      CREATE PROCEDURE GETFILE (
+      IN FILENAME VARCHAR(256) CHARACTER SET ISO88591,
+      IN OFFSET INTEGER,
+      OUT FILEDATA VARCHAR(102400) CHARACTER SET ISO88591,
+      OUT DATALENGTH LARGEINT)
+      EXTERNAL NAME 'com.traf.mgmt.JarFileMgmt.get (java.lang.String,int,java.lang.String[],long[])'
+      EXTERNAL SECURITY DEFINER
+      LIBRARY SPJMGMT
+      LANGUAGE JAVA
+      PARAMETER STYLE JAVA
+      READS SQL DATA
+      ;
+      GRANT EXECUTE ON PROCEDURE GETFILE TO PUBLIC;
+      
+      cqd CAT_IGNORE_ALREADY_EXISTS_ERROR 'off';
+      cqd CAT_IGNORE_DOES_NOT_EXIST_ERROR 'off';
+
+      exit;
+procEOF
+    
+}
+
+dropAndCreateSchema
+createProcedures
+echo "Successfully completed installation of SPJs."
```

Comment:
We do have special roles today that we use to own external schemas for HIVE and HBASE.  They are called DB__HIVEROLE